src/ChangeStream.php

```diff
@@ -60,7 +62,7 @@ public function next()
     {
         try {
             $this->csIt->next();
-            $this->resumeToken = $this->extractResumeToken($this->csIt->current());
+            $this->extractResumeToken($this->csIt->current());
```

Comment:
Maybe wrap this in `if ($this->valid())` in case `next()` just did a getMore but there were no new results.

Reply:
Typically, `$this->valid()` would check if `$this->key() !== null`. That still works, provided we're actually calling `key()` on the internal iterator.

As for `$this->key()`, we can't simply call `key()` on the internal iterator because resuming ends up recreating a PHP cursor, which would reset it's key sequence at zero. Therefore, you probably need to keep track of an integer sequence yourself.